tests/rptest/tests/transactions_test.py

```diff
@@ -489,6 +490,56 @@ def consume_records(consumer, count):
 
         consume_records(consumer, 200)
 
+    @cluster(num_nodes=3)
+    def check_pids_overflow_test(self):
```

Comment:
we also need a test to check that the fresh producers still work

Reply:
we check that 0th producer doesn't work after 50 new producers, but imagine that `max_concurrent_producer_ids` is hard-coded to one while we try to set it to 10 - this test won't catch this. please check that 41th producer still work after we used 50th producer.